Explore Datatable crash when dimension is empty

Explore UI errors out when a time-series based chart is selected and one of the non-time dimensions has an empty value `''`.

#### How to reproduce the bug

1. Load the `birth_names` example dataset
2. Add a calculated column called "empty" with the following expression: `CASE WHEN gender = 'girl' THEN 'Girl' ELSE '' END`
3. Create a chart based on the `birth_names`
4. Select any time-series viz type
5. Group by the new calculated dimension
6. Update Chart

### Expected results

Results tab loads the results grouped by time with the calculated dimension as columns.

### Actual results

Page goes blank with the following error:
```
react-dom.production.min.js:209 Error: A column ID (or string accessor) is required!
    at x (react-table.production.min.js:1:4204)
    at react-table.production.min.js:1:3337
    at Array.map (<anonymous>)
    at w (react-table.production.min.js:1:3313)
    at react-table.production.min.js:1:53001
    at Object.useMemo (react-dom.production.min.js:165:493)
    at t.useMemo (react.production.min.js:25:113)
    at e.useTable (react-table.production.min.js:1:52974)
    at b (TableView.tsx:151:7)
    at Ko (react-dom.production.min.js:153:146)
```

### Environment

- superset version: 8d4994a8 (current `master`) and also 1.5.0

### Checklist

Make sure to follow these steps before submitting your issue - thank you!

- [x] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [x] I have reproduced the issue with at least the latest released version of superset.
- [x] I have checked the issue tracker for the same issue and I haven't found one similar.

### Additional context

Reintroduced with https://github.com/apache/superset/pull/18569, previously fixed with https://github.com/apache/superset/pull/17303

--- a/superset-frontend/src/explore/components/DataTableControl/index.tsx
+++ b/superset-frontend/src/explore/components/DataTableControl/index.tsx
@@ -314,9 +314,9 @@ export const useTableColumns = (
               const isOriginalTimeColumn =
                 originalFormattedTimeColumns.includes(key);
               return {
-                id: key,
+                // react-table requires a non-empty id, therefore we introduce a fallback value in case the key is empty
+                id: key || index,
                 accessor: row => row[key],
-                // When the key is empty, have to give a string of length greater than 0
                 Header:
                   colType === GenericDataType.TEMPORAL &&
                   typeof firstValue !== 'string' ? (

